app/src/main/java/com/mapbox/navigation/examples/junctions/ShowJunctionsActivity.kt

```diff
@@ -0,0 +1,238 @@
+            }
+            startSimulation(result.navigationRoutes[0])
+        }
+    }
+
+    private val bannerInstructionsObserver = BannerInstructionsObserver { bannerInstructions ->
+        // The junction component is driven by banner instructions updates.
+        // Passing the instructions to the MapboxJunctionApi generates the data
+        // for updating the view.
+        junctionApi.generateJunction(bannerInstructions, junctionCallback)
+    }
+
+    private fun init() {
+        initNavigation()
+        initStyle()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initNavigation() {
+        mapboxNavigation = MapboxNavigationProvider.create(
+            NavigationOptions.Builder(this)
+                .accessToken(getString(R.string.mapbox_access_token))
+                .locationEngine(ReplayLocationEngine(mapboxReplayer))
+                .build()
+        )
+        mapboxNavigation.startTripSession()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initStyle() {
+        mapboxMap.loadStyleUri(Style.MAPBOX_STREETS) { style ->
+            routeLineView.initializeLayers(style)
+            binding.actionButton.setOnClickListener {
+                mapboxNavigation.setNavigationRoutes(listOf(route))
+                binding.actionButton.visibility = GONE
+            }
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun startSimulation(route: NavigationRoute) {
+        mapboxReplayer.stop()
+        mapboxReplayer.clearEvents()
+        mapboxReplayer.pushRealLocation(this, 0.0)
+        val replayEvents = ReplayRouteMapper().mapDirectionsRouteGeometry(route.directionsRoute)
+        mapboxReplayer.pushEvents(replayEvents)
+        mapboxReplayer.seekTo(replayEvents.first())
+        mapboxReplayer.play()
+    }
+
+    private fun updateCamera(location: Location) {
+        val mapAnimationOptionsBuilder = MapAnimationOptions.Builder()
+        mapAnimationOptionsBuilder.duration(1500L)
+        binding.mapView.camera.easeTo(
+            CameraOptions.Builder()
+                .center(Point.fromLngLat(location.longitude, location.latitude))
+                .bearing(location.bearing.toDouble())
+                .pitch(45.0)
+                .zoom(17.0)
+                .padding(EdgeInsets(1000.0, 0.0, 0.0, 0.0))
+                .build(),
+            mapAnimationOptionsBuilder.build()
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = MapboxActivityShowJunctionsBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        mapboxMap = binding.mapView.getMapboxMap()
+        locationComponent = binding.mapView.location.apply {
+            setLocationProvider(navigationLocationProvider)
+            enabled = true
+        }
+        init()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        if (::mapboxNavigation.isInitialized) {
```

Comment:
This check is not necessary, because `mapboxNavigation` is initialized in `onCreate`. The same with `onStop`.